### ISSUE TITLE
Using nbclient's async context manager to initialize kernel client

### ISSUE THREAD

**Issue by SylvainCorlay** [voila-dashboards/voila#591]
In `async def _jinja_kernel_start`, we [currently do](https://github.com/voila-dashboards/voila/blob/master/voila/handler.py#L126-L138)

```
        self.executor.kc = km.client()
        self.executor.kc.start_channels()
        try:
            await self.executor.kc.wait_for_ready(timeout=self.executor.startup_timeout)
        except RuntimeError:
            self.executor.kc.stop_channels()
            self.executor.km.shutdown_kernel()
            raise
```

I think it may be worth looking into nbclient's async context manager to setup kernel which appears to be doing the same thing.

cc @davidbrochart @maartenbreddels.

**Comment by maartenbreddels** [comment]
Totally agree, I had to do a similar dance for the widget testing framework, I think as a library (nbclient) we should make it easier.

**Comment by davidbrochart** [comment]
I don't think we can use nbclient's async context manager as it also takes care of the client termination, and we need the client to stay alive between `_jinja_kernel_start` and `_jinja_cell_generator`. But I have opened a PR to make it possible to use `(async_)start_new_kernel_client`: https://github.com/jupyter/nbclient/pull/51
